### PR TITLE
Add integration with Hugging Face

### PR DESCRIPTION
**Pull request by NielsRogge**
Hi,

Thanks for this nice work. I wrote a quick PoC to showcase that you can easily have integration so that you can automatically load the various DDColor models using `from_pretrained` (and push them using `push_to_hub`), track download numbers for your models (similar to models in the Transformers library), and have nice model cards on a per-model basis. It leverages the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which allows to inherits these methods.

This allows for:

```
from basicsr.archs.ddcolor_arch import DDColorHF

model = DDColorHF.from_pretrained("nielsr/ddcolor-tiny")
```
Importantly is that you will see download numbers of your models similar to all models on the 🤗  hub, on a per-checkpoint basis. Hence this can be done for each model in your [model zoo](https://github.com/piddnad/DDColor/blob/master/MODEL_ZOO.md).

We could move all checkpoints to separate repos on the [Alibaba organization](https://huggingface.co/alibaba-pai) if you're interested.

I saw that currently download numbers won't be tracked since they don't leverage this integration.

Would you be interested in this integration?

Kind regards,

Niels

### CODE REVIEW

**Comment by piddnad** [comment]
Hi @NielsRogge, thank you for your contribution! I will provide a review for this PR shortly.

File: basicsr/archs/ddcolor_arch.py [hunk 1]
```diff
@@ -383,3 +385,34 @@ def forward(self, img_features, encode_feat):
...
+    encoder_name='convnext-t',
+    decoder_name='MultiScaleColorDecoder',
+    input_size=[512, 512],
+    num_output_channels=2,
+    last_norm='Spectral',
+    do_normalize=False,
+    num_queries=100,
+    num_scales=3,
+    dec_layers=9,
+)
```
Comment (piddnad) [inline basicsr/archs/ddcolor_arch.py:407]: This example could be added to the README.md, perhaps by adding a new section "Inference with HuggingFace" under "Quick Start". In any case, I don't think it should be included within the package itself.

**Comment by piddnad** [comment]
Hi @NielsRogge , I have added the comment. 

Additionally, there are two questions I would like to know:
1. After the integration, is it necessary to create a separate HuggingFace repository for each different model, e.g., ddcolor-paper, ddcolor-tiny, etc.?
2. Do I need to upload the models myself? If you could help me move all the checkpoints to separate repositories, that would be greatly appreciated. For now, it's fine to keep the models in my own HuggingFace account repository.

Thank you for your time and assistance.

**Comment by NielsRogge** [comment]
Hi @piddnad,

Thanks for your review! 

> 1. After the integration, is it necessary to create a separate HuggingFace repository for each different model, e.g., ddcolor-paper, ddcolor-tiny, etc.?

Yes, with this Mixin class you can have download metrics per model. Hence we would have a separate Hugging Face repository for each different checkpoint (similar to how this is the case for models in the Transformers/Diffusers libraries).

> Do I need to upload the models myself? If you could help me move all the checkpoints to separate repositories, that would be greatly appreciated. For now, it's fine to keep the models in my own HuggingFace account repository.

I have updated the conversion script, I was wondering whether you would be willing to run `python basicsr/archs/ddcolor_arch.py` from my branch (which pushes each checkpoint to the hub). I got 2 unexpected keys when loading the weights, `["encoder.arch.head_cls.weight", "encoder.arch.head_cls.bias"]`. As you are more familiar with the model, you might be better suited to run the conversion script.

Kind regards,

Niels

**Comment by piddnad** [comment]
Hi @NielsRogge , I have added integration with Hugging Face. 

Thank you for your suggestion and example code!

**Comment by NielsRogge** [comment]
Awesome, thanks so much!

**Comment by NielsRogge** [comment]
I've added the "ddcolor" tag in the READMEs: https://huggingface.co/models?other=ddcolor

Looks like download numbers are working :) will close this PR.

**Pull request closed without merging**